src/models.ts

```diff
@@ -67,120 +68,202 @@ export enum SortField {
-  alltimeview: 'desc',
-  weeklyview: 'desc',
-  title: 'asc',
-  date: 'desc',
-  datearchived: 'desc',
-  datereviewed: 'desc',
-  dateadded: 'desc',
-  creator: 'asc',
-};
-
-/**
- * Maps the SortField above to the corresponding Metadata field in the API.
- */
-export const SortFieldToMetadataField: {
-  [key in SortField]: MetadataSortField | null;
-} = {
-  default: null,
-  relevance: null,
-  alltimeview: 'downloads',
-  weeklyview: 'week',
-  title: 'titleSorter',
-  date: 'date',
-  datearchived: 'publicdate',
-  datereviewed: 'reviewdate',
-  dateadded: 'addeddate',
-  creator: 'creatorSorter',
-};
-
-/**
- * Maps the Metadata field to the corresponding SortField field in the API.
- */
-export const MetadataFieldToSortField: {
-  [key in MetadataSortField]: SortField;
-} = {
-  week: SortField.weeklyview,
-  downloads: SortField.alltimeview,
-  titleSorter: SortField.title,
-  date: SortField.date,
-  publicdate: SortField.datearchived,
-  reviewdate: SortField.datereviewed,
-  addeddate: SortField.dateadded,
-  creatorSorter: SortField.creator,
-};
+export interface SortOption {
+  /**
+   * The SortField enum member corresponding to this option.
+   */
+  field: SortField;
+
+  /**
+   * The default sort direction to apply when this sort option is first selected.
+   */
+  defaultSortDirection: SortDirection | null;
+
+  /**
+   * Whether this sort option allows its sort direction to be changed from the default.
+   */
+  canSetDirection: boolean;
+
+  /**
+   * Whether this sort option may appear in the sort bar.
+   */
+  shownInSortBar: boolean;
+
+  /**
+   * Whether this sort option should be saved to the URL.
+   * If false, then no `sort` param will be added to the URL when this sort option
+   * is selected.
+   */
+  shownInURL: boolean;
+
+  /**
+   * Whether this sort option is passed to the search service.
+   * If false, then no sort param will be passed to the search service at all when
+   * this sort option is selected.
+   */
+  handledBySearchService: boolean;
+
+  /**
+   * The string identifying this sort field to the search service & backend API.
+   */
+  searchServiceKey?: string;
+
+  /**
+   * The human-readable name to use for this option in the sort bar (if applicable).
+   */
+  displayName: string;
+
+  /**
+   * A list of URL param keys that should be mapped to this sort option.
+   * E.g., both `title` and `titleSorter` in the URL map to the `SortField.title` option.
+   */
+  urlNames: (string | null | undefined)[];
+}
 
-/**
- * Maps the Metadata fields to how they should be presented in URLs.
- */
-export const MetadataFieldToURLField: {
-  [key in MetadataSortField]: URLSortField;
-} = {
-  week: 'week',
-  downloads: 'downloads',
-  titleSorter: 'title',
-  date: 'date',
-  publicdate: 'publicdate',
-  reviewdate: 'reviewdate',
-  addeddate: 'addeddate',
-  creatorSorter: 'creator',
+export const SORT_OPTIONS: Record<SortField, SortOption> = {
```

Comment:
This map consolidates all the properties of each sort field into a single config object, so that we don't have to hunt them down across a bunch of other maps. This should hopefully be easier & more organized than before.

In particular:
- the previous `SortFieldDisplayName` map became the `displayName` property.
- the previous `DefaultSortDirection` map became the `defaultSortDirection` property.
- the previous `MetadataFieldToSortField`, `MetadataFieldToURLField`, and `URLFieldToSortField` maps are now all handled by the `urlNames` property, which serves to list all the valid aliases for a given sort option that are recognized from the URL or backend APIs.
- the previous `SortFieldToMetadataField` map is now handled by the `searchServiceKey` property, indicating the correct key to request that sort from the search service.
- we add the `canSetDirection`, `shownInSortBar`, `shownInURL`, and `handledBySearchService` properties as a more declarative alternative to some of the previous sort bar & URL logic that addressed these as special cases.